crates/cast/src/starknet_commands/account/add.rs

```diff
@@ -1,48 +1,63 @@
 use crate::starknet_commands::account::{prepare_account_json, write_account_to_accounts_file};
-use anyhow::{ensure, Result};
+use anyhow::{anyhow, bail, ensure, Context, Result};
 use camino::Utf8PathBuf;
 use cast::get_chain_id;
 use cast::helpers::response_structs::AccountAddResponse;
 use cast::helpers::scarb_utils::CastConfig;
+use cast::{get_keystore_password, parse_number};
 use clap::Args;
-use starknet::core::types::BlockTag::Latest;
+use starknet::core::types::BlockTag::Pending;
 use starknet::core::types::{BlockId, FieldElement};
+use starknet::core::utils::get_contract_address;
 use starknet::providers::{
     jsonrpc::{HttpTransport, JsonRpcClient},
     Provider,
 };
 use starknet::signers::SigningKey;
+use std::fs;
 
-#[derive(Args, Debug)]
+#[derive(Args, Debug, Default)]
 #[command(about = "Add an account to the accounts file")]
 pub struct Add {
     /// Name of the account to be added
     #[clap(short, long)]
     pub name: String,
 
     /// Address of the account
-    #[clap(long, visible_alias = "addr")]
+    #[clap(
+        visible_alias = "addr",
+        long,
+        default_value = "0",
+        required_unless_present = "keystore",
+        conflicts_with = "keystore"
+    )]
     pub address: FieldElement,
 
     /// Class hash of the account
-    #[clap(short, long)]
+    #[clap(short, long, conflicts_with = "keystore")]
     pub class_hash: Option<FieldElement>,
 
     /// Account deployment status
     /// If not passed, sncast will check whether the account is deployed or not
-    #[clap(short, long)]
+    #[clap(short, long, conflicts_with = "keystore")]
     pub deployed: bool,
 
     /// Account private key
-    #[clap(long)]
+    #[clap(
+        long,
+        visible_alias = "priv",
+        default_value = "0",
+        required_unless_present = "keystore",
+        conflicts_with = "keystore"
+    )]
     pub private_key: FieldElement,
 
     /// Account public key
-    #[clap(long)]
+    #[clap(long, visible_alias = "pub")]
     pub public_key: Option<FieldElement>,
```

Comment:
in your examples in description I can see that public key is included in starkli account file. Wouldn't it make sense to also set is as conflicted with keystore, and just take what's in the file?